src/richie/apps/courses/models/enrollment.py

```diff
@@ -0,0 +1,70 @@
+"""
+Declare and configure `courses` application models related to enrollment.
+"""
+from django.contrib.auth import get_user_model
+from django.db import models
+from django.utils.translation import gettext_lazy as _
+
+from .course import CourseRun
+
+
+class Enrollment(models.Model):
+    """
+    Enrollment model. Links a user to the course runs they have enrolled in.
+
+    In most cases, actual enrollment also needs to happen in the actual LMS behind Richie.
+    Still, we need a simple and fast way to look up and manage user enrollments in
+    Richie itself, which is tough to achieve if every action necessitates a bunch of
+    API calls to different LMSes.
+
+    This is where the Enrollment model comes in: it is kept in sync with enrollments in various
+    LMSes but provides a unified Django way to know about them and manage them.
+    """
+
+    REQUESTED = "requested", _("requested")
+    ENROLLED = "enrolled", _("enrolled")
+    FAILED = "failed", _("failed")
+    STATE_CHOICES = [REQUESTED, ENROLLED, FAILED]
+
+    created_at = models.DateTimeField(verbose_name=_("created at"), auto_now_add=True)
+
+    user = models.ForeignKey(
+        verbose_name=_("user"),
+        help_text=_("user whose enrollment is represented"),
+        to=get_user_model(),
+        on_delete=models.CASCADE,
+        related_name="enrollments",
+        related_query_name="enrollment",
+    )
+    course_run = models.ForeignKey(
+        verbose_name=_("course run"),
+        help_text=_("course run the user enrolled in"),
+        to=CourseRun,
+        on_delete=models.CASCADE,
+        related_name="+",
```

Comment:
could be interesting to get the users enrolled on a course run? course_run.users_enrolled?